shoop/xtheme/views/extra.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import six
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.signals import setting_changed
+from django.http.response import HttpResponseNotFound
+
+from shoop.utils.importing import load
+from shoop.xtheme.theme import get_current_theme
+
+_VIEW_CACHE = {}
+
+
+def clear_view_cache(**kwargs):
+    _VIEW_CACHE.clear()
+
+
+setting_changed.connect(clear_view_cache, dispatch_uid="shoop.xtheme.views.extra.clear_view_cache")
+
+
+def _get_view_by_name(theme, view_name):
+    view = getattr(theme, "views", {}).get(view_name)
+    if isinstance(view, six.string_types):  # Handle loading views from specs
+        try:
+            view = load(view)
+        except ImproperlyConfigured:
+            if settings.DEBUG:
+                raise
+            return None
```

Comment:
If theme's views dict has the views as objects rather than strings, it makes it easier to navigate from the theme class to the views by go-to-definition feature of IDE, since that works better for objects than for strings. That's also where Django is heading with its urls stuff: https://docs.djangoproject.com/en/dev/releases/1.8/#django-conf-urls-patterns